ocaml-lsp-server/test/e2e-new/document_flow.ml

```diff
@@ -0,0 +1,55 @@
+open Test.Import
+
+let%expect_test "double open a document" =
+  let diagnostics = Fiber.Mvar.create () in
```

Comment:
an idea to consider in the future:
I've looked up an implementation of code_action.ml from e2e-new and the test environment setup looks pretty similar. I wonder if we can abstract this part so that we can write less boilerplate.

Reply:
if we improve our client implementation, i would imagine some of this code to be simpler.